piet-cairo/src/text/lines.rs

```diff
@@ -132,10 +132,17 @@ pub(crate) fn calculate_line_metrics(text: &str, font: &ScaledFont, width: f64)
             }
 
             // now do the hard break
+            let mut line_end = line_break;
+            // Remove any trailing newline characters from the line.
+            while matches!(text.get(line_end - 1..line_end), Some("\n") | Some("\r"))
```

Comment:
Was slightly confused here for a second; I couldn't tell that this wouldn't screw up in cases where there were multiple newlines in a row, which should be treated as actual newlines.

Since this is linux I'm also not sure we care about `\r`?

just for fun I'd have probably written something like,

```rust
let line_text = &text[line_start..line_end];
let newline_len = line_text.as_bytes().rev().take_while(|b| b == b'\n'').count();
let line_end = line_end - newline_len;
```

Reply:
Yeah `\r` is probably overkill.

Reply:
I also think your solution is better.